internal/controllers/mic_reconciler.go

```diff
@@ -0,0 +1,336 @@
+				err := mrhi.podHelper.createPullPod(ctx, &imageSpec, micObj)
+				errs = append(errs, err)
+			}
+		case kmmv1beta1.ImageDoesNotExist:
+			if imageSpec.Build == nil && imageSpec.Sign == nil {
+				break
+			}
+			fallthrough
+		case kmmv1beta1.ImageNeedsBuilding:
+			// image needs to be built - patching MBSC
+			mbscObj := &kmmv1beta1.ModuleBuildSignConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: micObj.Name, Namespace: micObj.Namespace},
+			}
+			_, err := controllerutil.CreateOrPatch(ctx, mrhi.client, mbscObj, func() error {
+				mrhi.mbscHelper.SetModuleImageSpec(mbscObj, &imageSpec)
+				mbscObj.Spec.ImageRepoSecret = micObj.Spec.ImageRepoSecret
+				return controllerutil.SetOwnerReference(micObj, mbscObj, mrhi.scheme)
+			})
+			errs = append(errs, err)
+		}
+	}
+	return errors.Join(errs...)
+}
+
+//go:generate mockgen -source=mic_reconciler.go -package=controllers -destination=mock_mic_reconciler.go pullPodManager
+type pullPodManager interface {
+	listImagesPullPods(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig) ([]v1.Pod, error)
+	deletePod(ctx context.Context, pod *v1.Pod) error
+	createPullPod(ctx context.Context, imageSpec *kmmv1beta1.ModuleImageSpec, micObj *kmmv1beta1.ModuleImagesConfig) error
+	getPullPodForImage(pods []v1.Pod, image string) *v1.Pod
+}
+
+type pullPodManagerImpl struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func newPullPodManager(client client.Client, scheme *runtime.Scheme) pullPodManager {
+	return &pullPodManagerImpl{
+		client: client,
+		scheme: scheme,
+	}
+}
+
+func (ppmi *pullPodManagerImpl) listImagesPullPods(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig) ([]v1.Pod, error) {
+	logger := ctrl.LoggerFrom(ctx).WithValues("mic name", micObj.Name)
+
+	pl := v1.PodList{}
+
+	hl := client.HasLabels{imageLabelKey}
+	ml := client.MatchingLabels{moduleImageLabelKey: micObj.Name}
+
+	logger.V(1).Info("Listing mic image Pods")
+
+	if err := ppmi.client.List(ctx, &pl, client.InNamespace(micObj.Namespace), hl, ml); err != nil {
+		return nil, fmt.Errorf("could not list mic image pods for mic %s: %v", micObj.Name, err)
+	}
+
+	return pl.Items, nil
+}
+
+func (ppmi *pullPodManagerImpl) deletePod(ctx context.Context, pod *v1.Pod) error {
+	logger := ctrl.LoggerFrom(ctx)
+	if pod.DeletionTimestamp != nil {
+		logger.Info("DeletionTimestamp set, pod is already in deletion", "pod", pod.Name)
+		return nil
+	}
+	if err := ppmi.client.Delete(ctx, pod); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to delete pull pod %s/%s: %v", pod.Namespace, pod.Name, err)
+	}
+	return nil
+}
+
+func (ppmi *pullPodManagerImpl) createPullPod(ctx context.Context, imageSpec *kmmv1beta1.ModuleImageSpec, micObj *kmmv1beta1.ModuleImagesConfig) error {
+	restartPolicy := v1.RestartPolicyOnFailure
+	if imageSpec.Build != nil || imageSpec.Sign != nil {
+		restartPolicy = v1.RestartPolicyNever
+	}
+	imagePullSecrets := []v1.LocalObjectReference{}
```

Comment:
```suggestion
	var imagePullSecrets []v1.LocalObjectReference
```
There is no point of initializing it if we override it [here](https://github.com/kubernetes-sigs/kernel-module-management/pull/1002/files#diff-9101b15d22c840a3bae15419ccf8c1aaaae06da4d9d15f59e798045d152e2866R296).

Reply:
fixed